lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,360 @@
+        socket.setNoDelay();
+        socket.setKeepAlive(true);
+        socket.unref();
+        return socket;
+    }
+
+    _propError(pdu) {
+        const statusCode = pdu.getStatusCode();
+        if (statusCode === kineticlib.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        if (statusCode !== kineticlib.errors.SUCCESS) {
+            return pdu.getErrorMessage();
+        }
+        return undefined;
+    }
+
+    /**
+     * set the sequence of the request.
+     *
+     * @param {number} index - the index of the socket.
+     * @param {number} sequence - the sequence to be set.
+     * @returns {Kinetic} - To allow for a functional style.
+     */
+    setSequence(index, sequence) {
+        if (sequence >= Number.MAX_VALUE) {
+            const temp = this._sockets[index].sock;
+            this._sockets[index].sock = this._newSocket();
+            this._sockets[index].sock.on('connect', () => {
+                this._bindReadable(this._sockets[index], index);
+            }).connect(this._sockets[index].drive);
+            temp.destroy();
+            this._sockets[index].sequence = 0;
+        } else {
+            this._sequence = sequence;
+            this._sockets[index].sequence = sequence;
+        }
+        return this;
+    }
+
+    /**
+     * get the pool of socket.
+     *
+     * @returns {Kinetic._sockets} - the pool of socket.
+     */
+    getSockets() {
+        return this._sockets;
+    }
+
+    /**
+     * get the socket at the index.
+     *
+     * @param {number} index - the index of the socket.
+     * @returns {Object} - the socket.
+     */
+    getSocket(index) {
+        return this._sockets[index];
+    }
+
+    /**
+     * get index of the socket.
+     *
+     * @param {string} host - the host of the socket.
+     * @param {string} port - the port of the socket.
+     * @returns {number|object} - the index of the socket or null if it does not
+     *                            exist.
+     */
+    getSocketIndex(host, port) {
+        this._sockets.forEach((socket, index) => {
+            if (socket.host === host && socket.port === port) {
+                return index;
+            }
+            if (index === this._sockets.length - 1) {
+                return null;
+            }
+            return undefined;
+        });
+    }
+
+    _incrString(str) {
+        this._lexiKey = (parseInt(str, 36) + 1).toString(36);
+        return this._lexiKey;
+    }
+
+    /**
+     * send the put request to the right drive.
+     *
+     * @param {number} index - the index of the socket
+     * @param {Buffer} value - the value to be put.
+     * @param {Object} options - the options object for the put request:
+     *                     kinetic-protocol
+     *        {string} options.synchronization - synchronization mode
+     *        {boolean} options.force - option for forcing the put
+     * @param {function} callback - the callback
```

Comment:
I would love to see the callback arguments defined, and the same goes for every other callback.